src/client/client.cpp

```diff
@@ -21,42 +21,30 @@ using namespace apache::thrift::transport;
 
 class ClientHandler {
   private:
-    std::ifstream seed_data;
-    bool init_db = false;
-    int num_clients = 16;
-    float p_get = 0.5;
-
-    std::vector<float> latencies;
+    ClientConfig config;
+    std::vector<double> latencies;
 
   public:
-    ClientHandler(int argc, char *argv[]) {
-        parseArgs(argc, argv, seed_data, init_db, num_clients, p_get);
-    }
+    ClientHandler(int argc, char *argv[]) { parseArgs(argc, argv, config); }
 
-    void start() {
-        if (init_db) {
-            initDB();
-        } else {
-            runThreaded();
-        }
-    }
+    void start() { (config.init_db) ? initDB() : runThreaded(); }
 
     void initDB() {
         redisCli rd;
         auto pipeline = rd.pipe();
 
         // If seed data exists, initialize the db with seed data
-        if (seed_data.is_open()) {
+        if (config.seed_data.is_open()) {
             std::string line;
-            while (std::getline(seed_data, line)) {
-                Operation op = getSeedOperation(line);
+            while (std::getline(config.seed_data, line)) {
+                Operation op = getSeedOperation(config);
                 pipeline.set(op.key, op.value);
             }
         }
```

Comment:
This logic is broken... 

`getline(config.seed_data, line)` removes a line from the ifstream but then `line` is never used. Then, `getSeedOperation` calls `readFile` which acquires a lock on the seed file and also gets a new line...?

Reply:
Have you tested this branch with seed data?

Reply:
Also DO NOT PULL MAIN until you get the branch working lol, many many conflicts.

Reply:
Good catch, I was still able to run initdb with seed data but the values would definitely be incorrect. I’ll amend this in the morning!

Reply:
This should be fixed now